services/src/main/java/io/druid/cli/CliPeon.java

```diff
@@ -232,22 +232,22 @@ public void run()
       Injector injector = makeInjector();
       try {
         final Lifecycle lifecycle = initLifecycle(injector);
-        Runtime.getRuntime().addShutdownHook(
-            new Thread(
-                new Runnable()
-                {
-                  @Override
-                  public void run()
-                  {
-                    log.info("Running shutdown hook");
-                    lifecycle.stop();
-                  }
-                }
-            )
+        final Thread hook = new Thread(
+            new Runnable()
+            {
+              @Override
+              public void run()
+              {
+                log.info("Running shutdown hook");
+                lifecycle.stop();
+              }
+            }
         );
+        Runtime.getRuntime().addShutdownHook(hook);
         injector.getInstance(ExecutorLifecycle.class).join();
         // Explicitly call lifecycle stop, dont rely on shutdown hook.
         lifecycle.stop();
+        Runtime.getRuntime().removeShutdownHook(hook);
```

Comment:
I am not opposed to the changes in this file though, as it adds better future proofing of the shutdown process for peons